pkg/controllers/workspace_controller.go

```diff
@@ -168,25 +186,36 @@ func selectWorkspaceNodes(qualified []*corev1.Node, preferred []string, previous
 			} else if !iPrevious && jPrevious {
 				return false
 			} else { // either all are previous, or none is previous
-				_, iCreatedByKaito := qualified[i].Labels["kaito.sh/machine-type"]
-				_, jCreatedByKaito := qualified[j].Labels["kaito.sh/machine-type"]
-
+				var iCreatedByGPUProvisioner, jCreatedByGPUProvisioner bool
+				_, iCreatedByGPUProvisioner = qualified[i].Labels[machine.LabelGPUProvisionerCustom]
+				_, jCreatedByGPUProvisioner = qualified[j].Labels[machine.LabelGPUProvisionerCustom]
 				// Choose node created by gpu-provisioner since it is more likely to be empty to use.
-				if iCreatedByKaito && !jCreatedByKaito {
+				if iCreatedByGPUProvisioner && !jCreatedByGPUProvisioner {
 					return true
-				} else if !iCreatedByKaito && jCreatedByKaito {
+				} else if !iCreatedByGPUProvisioner && jCreatedByGPUProvisioner {
 					return false
 				} else {
-					return qualified[i].Name < qualified[j].Name
+					var iCreatedByKarpenter, jCreatedByKarpenter bool
```

Comment:
Guard this code with feature gate?

Reply:
they will be always nil as no nodeclaims are available but I'm ok to add feature gate.